documentation/client-side-forms/appendix-client-side-form-binding-reference.asciidoc

```diff
@@ -0,0 +1,348 @@
+---
+title: "Appendix: Client-Side Form Binding Reference"
+order: 100
+layout: page
+---
+
+ifdef::env-github[:outfilesuffix: .asciidoc]
+
+= Appendix: Client-Side Form Binding Reference
+
+The key concepts behind client-side form binding are: the `field` directive, the `Model`, the `Binder` and binder nodes.
+
+== The field directive
+
+Client-side form binding in Vaadin works together with LitElement web component library, and its underlying template rendering library `lit-html`.
+
+`field` is a `lit-html` directive that does the main job of binding the form field components in the template, namely:
+
+- sets the `name` attribute
+- implements two-way binding for the value state
+- sets the `required` boolean state
+- sets the boolean `invalid` state and the `errorMessage` string state when the current value is invalid
+
+.Using the field directive
+[source, html]
+----
+<vaadin-text-field
+ label="Full name"
+ ...="${field(model.fullName)}"
+ ></vaadin-text-field>
+----
+
+Depending on the type of the bound component, the field directive selects a strategy, which defines how exacly the states above are applied on the component, e. g., which attributes and properties of the element are used.
+
+[NOTE]
+You can find more information on field strategy customization in the <<appendix-web-component-field-strategy#, Appendix: Using a Web Component Field>> article.
+
+The field directive supports Vaadin components and HTML input elements. While Vaadin components have support for all the states, for HTML input elements the `invalid`, `required` and `errorMessage` are not displayed in the bound component. As a workaround, you can bind those manually in the template:
+
+[source, html]
+----
+<label for="fullName">
+  Full name
+  ${binder.for(binder.model.fullName).required ? '*' : ''}
+</label>
+<input id="fullName" ...="${field(binder.model.fullName)}"/><br/>
+${
+  binder.for(binder.model.fullName).invalid
+    ? html`
+      <strong>
+        ${binder.for(binder.model.fullName).errors[0]}
+      </strong>`
+    : ''
+}
+----
+
+[NOTE]
+See also: <<appendix-vaadin-components.asciidoc#, Appendix: Binding Data to Vaadin Components>>
+
+== Model
+
+The model is a type-safe reference to a value stored in the binder. Models describe the structure of the data, and are used to read and modify current form values and other related properties and states.
```

Comment:
Style (minor): a general recommendation is that each sentence is written on a separate line. Makes it easier to review changes and change the order of sentences when editing.
```suggestion
The model is a type-safe reference to a value stored in the binder. 
Models describe the structure of the data, and are used to read and modify current form values and other related properties and states.
```